

# Divergence Time Inference with BEAST 2

This tutorial is modified from https://github.com/leimurillo/tutorials/tree/master/bayesian_phylogeny_inference.

Bayesian inference differs from Maximum Likelihood becase take into account prior probabilities. In the context of phylogenetic inference, this means that parameters could be constrained towards values (from fossils or biogeographic evidence) that are considered realistic based on the findings of previous studies on the group of interest. These in turn can then inform the timing in other parts of the phylogeny so that an overall timeline of diversification can be estimated. 

In this tutorial, we will demonstrate how time-calibrated phylogenies can be inferred with programs of the Bayesian software package BEAST2 (Bouckaert et al. 2014). BEAST2 is a cross-platform program for Bayesian phylogenetic analysis of molecular sequences. It estimates rooted, time-measured phylogenies using strict or relaxed molecular clock models. It can be used as a method of reconstructing phylogenies but is also a framework for testing evolutionary hypotheses without conditioning on a single tree topology. BEAST 2 uses Markov chain Monte Carlo (MCMC) to average over tree space, so that each tree is weighted proportional to its posterior probability. BEAST 2 includes a graphical user-interface for setting up standard analyses and a suit of programs for analysing the results. The BEAST2 package, including BEAUti, BEAST2 itself, TreeAnnotator, and other tools can be downloaded from the BEAST2 [website](https://www.beast2.org).

Tracer: Just like BEAST2, Tracer is written in Java and should work on your system without problems. The program can be downloaded for Mac OS X, Linux, or Windows from https://github.com/beast-dev/tracer/releases. The file with the extension .dmg is for Mac OS X, the one with the extension .tgz is for Linux, and the Windows version is the file ending in .zip.

FigTree: If you followed tutorial Maximum-Likelihood Phylogenetic Inference, you should already have FigTree installed. If not, you will find executables for Mac OS X, Linux, and Windows on https://github.com/rambaut/figtree/releases.

The data used in this tutorial are the cleaned version of the alignments generated in the tutorial Multiple Sequence Alignment. Please download the data from [here](https://github.com/niklas-w/Molecular-systematics-course/blob/master/Dataset/6.Dataset.nex)

The first step will be to convert a NEXUS file with a DATA block into a BEAST XML input file. This is done using the program BEAUti (which stands for Bayesian Evolutionary Analysis Utility). This is a user-friendly program for setting the evolutionary model and options for the MCMC analysis. The second step is to actually run BEAST using the input file generated by BEAUTi, which contains the data, model and analysis settings. The final step is to explore the output of BEAST in order to diagnose problems and to summarize the results.

Open the program BEAUti from the BEAST2 package, and import the alignment. To do so, click "Import Alignment" from the "File" menu and select the dataset.nex or by clicking the "+ symbol" and "add aligment". The BEAUti window should then look as shown in the screenshot below.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast1.png" alt="Beast1" width="600"></p>

The BEAUti interface has six different tabs, of which (at the top of the window), the first one "Partitions" is currently selected. First need to specify settings regarding the partitioning in the currently open tab. 

Select all four rows at the same time and click on "Link Trees" near the top of the BEAUti window. This will force BEAST2 to use the same phylogeny for all partitions, which is equivalent to concatenating the sequences as we did for maximum-likelihood analyses with RAxML in tutorial Maximum-Likelihood Phylogenetic Inference. 

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast3.png" alt="Beast3" width="600"></p>

After clicking on "Link Trees", you should notice that all cells in the column to the right, under the heading "Tree" show the same value, "COI-begin", as in the screenshot below. This indicates that they now all share the same tree. You can edit the name and assign the name "tree" for instance.

Now with all rows at the same time we will click on "Link Clock Models". This means that the clock model that we will  apply to all genes. With the relaxed clock model that we will select, it means that some branches are allowed to evolve faster than other branches, but that this variation in rates is not inferred separately for each gene. After clicking on "Link Clock Models", the BEAUti window should look as shown in the screenshot below. You can also edit the name and  assign the name "clock" for instance.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast4.png" alt="Beast4" width="600"></p>

The settings in the "Partitions" tab are now complete.

Click on the "Site Model" tab next. In this tab we can specify the substitution models for all our partitions. Select the COI-begin partition in the panel at the left and click on the drop-down menu that currently says "JC69". Instead of the Jukes-Cantor model, use the GTR model as suggested by the model selection. Also specify "4" in the field for the "Gamma Category Count" two lines above, to use a gamma model of rate variation with four rate categories. 

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast5.png" alt="Beast5" width="600"></p>

Still in the "Site Model" tab, select all three partitions in the panel at the left of the window. The main part of the window should then show the option "Clone from COI-begin" as in the screenshot below. Click "OK" to use the same site model for EF1a and wingless.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast6.png" alt="Beast6" width="600"></p>

Next, click on the "Clock Model" tab. From the drop-down menu that currently says "Strict Clock", choose "Relaxed Clock Log Normal" instead. This is the most commonly used relaxed clock model in which substitution rates of individual branches are drawn from a lognormal distribution.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast7.png" alt="Beast7" width="600"></p>

Click on the "Priors" tab. From the drop-down menu at the very top of the window, select "Birth Death Model" instead of "Yule Model". By doing so we add a parameter to the model for the extinction rate. If we would choose the alternative Yule model (Yule 1925), we would assume that no extinction have ever occurred. As this seems rather unrealistic, the birth-death model (Gernhard 2008) is in most cases the more appropriate choice. 

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast8.png" alt="Beast8" width="600"></p>

Most of the other items shown in the "Prior" panel correspond to prior densities placed on the parameters of the substitution models for the partitions. You may keep the default priors for each of these parameters. However, to allow time calibration of the phylogeny, a prior density still needs to be specified for at least one divergence time, otherwise BEAST2 would have very little information to estimate branch lengths according to an absolute time scale. 

Continue to the "MCMC" tab, where you can specify the run length. This analysis will require a few hundred million iterations before the MCMC chain reaches full stationarity, which would take several days of run time. For this exercise we recommend that you use a chain length of 5,000,000 million states and either run the analysis or cancel the run after following it for some time, and then use output files from our analysis (you'll find the links below) for the rest of the tutorial. 

Also change the names of the output files: Click on the triangle to the left of "tracelog" and specify "Dataset.log" as the name of the log file. In the next field for "Log Every", set the number to "1,000" so that only every 1,000 MCMC state is written to the log file. Click on the triangle again, then click on the black triangle to the left of "treelog". Specify "Dataset.trees" as the name of the tree file and again use "1,000" as the number in the field for "Log Every". When the window looks as in the below screenshot, click on "Save" in BEAUti's "File" menu, and name the resulting file in XML format Dataset.xml. You can download [Dataset.xml](https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Output/Dataset.xml) If you did not manage to complete this step.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast9.png" alt="Beast9" width="600"></p>

Now, open the program BEAST2 and select the file [Dataset.xml](https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Output/Dataset.xml) as input file, as shown in the screenshot below. When you click the "Run" button, BEAST2 will start the analysis.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast10.png" alt="Beast10" width="600"></p>

# Analyzing the results with Tracer

After the two BEAST2 analyses have completed (or if you decided not to wait and use [the output](https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Output/Dataset.log) of our analysis instead), open file [Dataset.log](https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Output/Dataset.log) in the program Tracer. When the main window has opened, choose Import Trace File from the File menu and select the file that BEAST has created called Dataset.log.

Remember that MCMC is a stochastic algorithm so the actual numbers will not be exactly the same as those depicted in the figure. On the left hand side is a list of the different quantities that BEAST has logged to file. There are traces for the posterior (this is the natural logarithm of the product of the tree likelihood and the prior density), and the continuous parameters. Selecting a trace on the left brings up analyses for this trace on the right hand side depending on tab that is selected. When first opened, the ‘posterior’ trace is selected and various statistics of this trace are shown under the Estimates tab. In the top right of the window is a table of calculated statistics for the selected trace.

In the bottom left part of the Tracer window, you'll see statistics for the estimate of the posterior probability (just named "posterior"), the likelihood, and the prior probability (just named "prior"), as well as for the parameters estimated during the analysis (except the phylogeny, which also represents a set of parameters). The second column in this part shows the mean estimates for each parameter and their ESS values. Do the ESS values of all parameters indicate stationarity?

With the posterior probability still being selected in the list at the bottom left, click on the tab for "Trace" (at the very top right). You will see how the posterior probability changed over the course of the MCMC. This trace plot should ideally have the form of a "hairy caterpillar", but as you can see from the next screenshot, this is not the case for the posterior probability.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast11.png" alt="Beast11" width="600"></p>


<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast12.png" alt="Beast12" width="600"></p>

# Obtaining the tree

BEAST also produces a posterior sample of phylogenetic time-trees along with its sample of parameter estimates. These need to be summarized using the program TreeAnnotator. This will take the set of trees and find the best supported one. It will then annotate this representative summary tree with the mean ages of all the nodes and the corresponding 95% HPD ranges. It will also calculate the posterior clade probability for each node. Run the TreeAnnotator program and set it up as depicted in Figure 

The burnin is the number of trees to remove from the start of the sample. Unlike Tracer which specifies the number of steps as a burnin, in Tree Annotator you need to specify the actual number of trees. For this run, you specified a chain length of 5,000,000 steps sampling every 1,000 steps. Thus the trees file will contain 5,000 trees and so to specify a 10% burnin in the top text field.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast13.png" alt="Beast13" width="600"></p>

For the input file, select the trees file that BEAST created and select a file for the output (here we called it [Final.tree](https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Output/Final.tree)). Now press Run and wait for the program to finish.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast14.png" alt="Beast14" width="600"></p>

Finally, we can visualize the tree in FigTree. Run this program, and open the Final.tree file by using the Open command in the File menu. The tree should appear.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast15.png" alt="Beast15" width="600"></p>

To add a time scale to the phylogeny, set a tick next to "Scale Axis" in the menu on the left. Also click the triangle next to it, remove the tick next to "Show grid" in the newly opened menu field, but set a tick next to "Reverse axis" instead. The result is shown in the next screenshot.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast16.png" alt="Beast16" width="600"></p>

To also see the confidence intervals for the age estimates, click on the triangle next to "Node Bars" in the menu on the left. Also set a tick in the checkbox for "Node Bars". Choose the first item from the drop-down menu for "Display", the "height_95%_HPD" The phylogeny should then look as shown below.

<p align="center"><img src="https://github.com/niklas-w/Molecular-systematics-course/blob/master/Tutorials/5.BayesianInference/Beast17.png" alt="Beast17" width="600"></p>
Question

1. Is this topology consistent with that recovered from IQTREE and RAxML?



